src/edu.umn.cs.melt.copper.compiletime/src/main/java/edu/umn/cs/melt/copper/compiletime/checkers/DisambiguationFunctionConflictChecker.java

```diff
@@ -0,0 +1,68 @@
+package edu.umn.cs.melt.copper.compiletime.checkers;
+
+import java.util.BitSet;
+import java.util.Hashtable;
+import java.util.Map.Entry;
+
+import edu.umn.cs.melt.copper.compiletime.auxiliary.SymbolTable;
+import edu.umn.cs.melt.copper.compiletime.logging.CompilerLogger;
+import edu.umn.cs.melt.copper.compiletime.logging.messages.DuplicateDisambiguationFunctionMessage;
+import edu.umn.cs.melt.copper.compiletime.logging.messages.OverlappingDisambiguationFunctionMessage;
+import edu.umn.cs.melt.copper.compiletime.spec.grammarbeans.CopperASTBean;
+import edu.umn.cs.melt.copper.compiletime.spec.numeric.ParserSpec;
+
+/**
+ * Checks that no two disambiguation functions have the same member set, and that all disambiguation
+ * functions applicable to subsets have disjoint member sets. 
+ * @author Lucas Kramer &lt;<a href="mailto:krame505@umn.edu">krame505@umn.edu</a>&gt;
+ *
+ */
+public class DisambiguationFunctionConflictChecker
+{
+	private CompilerLogger logger;
+	private SymbolTable<CopperASTBean> symbolTable;
+	private ParserSpec spec;
+	
+	private DisambiguationFunctionConflictChecker(CompilerLogger logger,SymbolTable<CopperASTBean> symbolTable,ParserSpec spec)
+	{
+		this.logger = logger;
+		this.symbolTable = symbolTable;
+		this.spec = spec;
+	}
+	
+	public static boolean check(CompilerLogger logger,SymbolTable<CopperASTBean> symbolTable, ParserSpec spec)
+	{
+		return new DisambiguationFunctionConflictChecker(logger, symbolTable, spec).checkDisambiguationFunctionConflicts();
+	}
+	
+	private boolean checkDisambiguationFunctionConflicts()
+	{
+		boolean passed = true;
+
+		Hashtable<BitSet,Integer> subsetDisambiguationFunctions = new Hashtable<BitSet,Integer>();
```

Comment:
Could improve speed here in the non-error case by comparing the sum of the cardinalities of all `appliesToSubsets` member sets and the cardinality of their union, which will be equal iff the sets are pairwise disjoint.